pkg/manifests/config.go

```diff
@@ -446,6 +427,35 @@ func (c *Config) LoadEnforcedBodySizeLimit(pcr PodCapacityReader, ctx context.Co
 	return poperator.ValidateSizeField(c.ClusterMonitoringConfiguration.PrometheusK8sConfig.EnforcedBodySizeLimit)
 }
 
+func (c *Config) precheck() error {
+	if c.ClusterMonitoringConfiguration.PrometheusK8sConfig.CollectionProfile != FullCollectionProfile && !c.TechPreview {
+		return errors.Wrap(ErrConfigValidation, "collectionProfiles is a TechPreview feature, to be able to use a profile different from the default (\"full\") please enable TechPreview")
+	}
+
+	// Validate the configured collection profile iff tech preview is enabled, even if the default profile is set.
+	if c.TechPreview {
+		for _, profile := range SupportedCollectionProfiles {
+			var v float64
+			if profile == c.ClusterMonitoringConfiguration.PrometheusK8sConfig.CollectionProfile {
+				v = 1
+			}
+			metrics.CollectionProfile.WithLabelValues(string(profile)).Set(v)
+		}
+		if !slices.Contains(SupportedCollectionProfiles, c.ClusterMonitoringConfiguration.PrometheusK8sConfig.CollectionProfile) {
+			return errors.Wrap(ErrConfigValidation, fmt.Sprintf(`%q is not supported, supported collection profiles are: %q`, c.ClusterMonitoringConfiguration.PrometheusK8sConfig.CollectionProfile, SupportedCollectionProfiles.String()))
+		}
+	}
+
+	var d float64
+	// Highlight deprecated config fields.
+	if c.ClusterMonitoringConfiguration.K8sPrometheusAdapter.DedicatedServiceMonitors != nil {
+		d = 1
+	}
+	// Deprecated in https://github.com/openshift/cluster-monitoring-operator/pull/2160
+	metrics.DeprecatedConfig.WithLabelValues("openshift-monitoring/cluster-monitoring-config", "k8sPrometheusAdapter.dedicatedServiceMonitors", "4.15").Set(d)
```

Comment:
Do we actually emit the 0 values? Instead we could just omit/remove the metric. That is the happy path and metric absence signals the same as the 0, but saves us a metric.

Reply:
yes, if the user gets rid of the deprecated config we need to stop emitting 1 (as in the test) and I don't think it's a good idea to just drop the metric, I don't think that`metrics` allows even doing that.
Plus it's good to be able to always list deprecated fields...

Reply:
fair point.